T2560 implement the write and pray feature

nfelber opened the pull request:
Depends on PR [T2602 Add already authenticated support to new sponsorship feature](https://github.com/CompassionCH/compassion-website/pull/133).

Implement Write&Pray page and wizard.

The workflow is the same as for regular sponsorships and shares most of the code, where Write&Pray sponsorships are treated as a special flavor of sponsorship.

### Images

<img width="815" height="855" alt="image" src="https://github.com/user-attachments/assets/edfe3d2b-b285-47af-a7ee-b7d3ba9287cb" />

<img width="815" height="855" alt="image" src="https://github.com/user-attachments/assets/1375c19c-e957-4c36-87fe-de522174ea6c" />

<img width="815" height="855" alt="image" src="https://github.com/user-attachments/assets/4070b17e-25f7-4f18-9f6d-9679e9ab9701" />

ecino reviewed (approved):
Check if Gemini's comment makes sense or not. Otherwise it looks good to me :+1:

Commit:
T2560 Implement the Write&Pray feature

# Conflicts:
#	my_compassion/controllers/my2_sponsorships.py
#	my_compassion/models/my2_new_sponsorship_wizard.py
#	my_compassion/static/src/js/my2_sponsorships.js
#	my_compassion/templates/pages/my2_new_sponsorship_wizard.xml
#	my_compassion/templates/pages/my2_sponsorships.xml

nfelber commented on my_compassion/models/my2_new_sponsorship_wizard.py:
```diff
@@ -82,16 +86,23 @@ def action_finish_sponsorship(self):
         # Set contract lines for the sponsorship
-        contract_lines = sponsorship._get_sponsorship_standard_lines(False)
+        contract_lines = sponsorship._get_sponsorship_standard_lines(self.write_and_pray)
         if not self.sponsorship_plus:
```
It is intended.
